sosia/establishing/logger.py

```diff
@@ -27,17 +28,25 @@ def create_logger(log_file: Union[str, Path] = None) -> None:
     logger.propagate = False
 
 
-def log_scopus(scopus_obj) -> None:
-    """Log the results of a Scopus query."""
-    scopus_class = scopus_obj.__class__.__name__
-    scopus_name = sub(r'(?<!^)([A-Z])', r' \1', scopus_class)
-    view = scopus_obj._view
-    query = scopus_obj._query
-
-    if logger is None:
-        create_logger()
-
-    logger.debug(
-        "\n\t- Scopus API: %s\n\t- View: %s\n\t- Query: %s\n\t",
-        scopus_name, view, query
-    )
+class ScopusLogger:
+    """Context manager to log scopus"""
+    def __init__(self, scopus_class_name, params):
+        self.scopus_obj: Union[AuthorSearch, ScopusSearch]
+        self.scopus_class_name = scopus_class_name
+        self.query = params.get('query')
+        self.view = params.get('view', 'Default')
+
+        if logger is None:
+            create_logger()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        if not exc_type:
+            results = self.scopus_obj.get_results_size()
+            self.view = self.scopus_obj._view
+        logger.debug(
+            '\n\t- Scopus API: %s with %s view \n\t- Query: %s\n\t- Results: %s',
+            self.scopus_class_name, self.view, self.query[:255], exc_type or results
```

Comment:
Please show the entire query - users are supposed to replicate the query if needed.